File: Python3/Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = {n: True}
        
        while (n is not 1):
            n = sum([int(c) ** 2 for c in str(n)])
            if n in seen:
                return False
            else:
                seen[n] = True
        return True

File: JavaScript/Happy Number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    while (n !== 1) {
        let x = 0
        for (let c of String(n)) {
            x += (Number(c) ** 2)
        }
        if (x === 4) return false
        n = x
    }
    return true
}
